src/components/Warning/Warning.tsx

```diff
@@ -0,0 +1,56 @@
+import { useEffect, useState } from 'react';
+import classNames from 'classnames';
+
+type Props = {
+  message: string;
+  errorCount: number;
+};
+
+const Warning: React.FC<Props> = ({ message, errorCount }) => {
+  const [isShown, setIsShown] = useState(false);
+
+  useEffect(() => {
+    let timer: ReturnType<typeof setTimeout>;
+
+    if (message) {
```

Comment:
use `useRef` to save timer

Reply:
use ref is for referencing certain html entities, not for timers, and I wrong?

Reply:
use ref is for referencing html entities not for timers. What do you mean by useRef?

Reply:
not only
refs are used every time you want to save any variable after re-renders